parlai/scripts/self_chat.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+"""
+Allows a model to self-chat on a given task.
+"""
+from parlai.core.params import ParlaiParser
+from parlai.core.agents import create_agent
+from parlai.core.worlds import create_task
+
+import random
+
+
+def setup_args(parser=None):
+    if parser is None:
+        parser = ParlaiParser(True, True, 'Self chat with a model')
+    parser.add_argument('-d', '--display-examples', type='bool', default=True)
+    parser.add_argument(
+        '--display-prettify',
+        type='bool',
+        default=False,
+        help='Set to use a prettytable when displaying '
+        'examples with text candidates',
+    )
+    parser.add_argument(
+        '--display-ignore-fields',
+        type=str,
+        default='label_candidates,text_candidates',
+        help='Do not display these fields',
+    )
+    parser.add_argument(
+        '-it',
+        '--interactive-task',
+        type='bool',
+        default=True,
+        help='Create interactive version of task',
+    )
+    parser.set_defaults(interactive_mode=True, task='self_chat')
```

Comment:
Do you need that task=self_chat?

Reply:
it makes it so that it can default do something, otherwise u need to specify the right task..